### DISCUSSION TITLE
How to implement a plugin for a custom language?

### DISCUSSION THREAD

**Discussion by demurgos** [yarnpkg/berry#4811] [Q&A]
Hello,
I would like to use Yarn to install packages for the Haxe language. Are there any examples in how to add support for a custom language?

The closest I found is [this MR to add CMake support](https://github.com/yarnpkg/berry/pull/1697) but it seems to patch a lot of the core files and it's hard to find what are core components for custom language support.

**Comment by RDIL** [comment]
It's worth noting that that PR is roughly 2 years old at this point.  The APIs have significantly improved since then.

To implement a new language as a plugin, you would need to add a fetcher and linker for the new language.  I believe there is documentation on the matter.
